File: fe2-android/app/src/test/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/Script_inputTest.java
package com.github.dedis.popstellar.model.network.method.message.data.digitalcash;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;

import org.junit.Test;

public class Script_inputTest {
  private static final String TYPE = "P2PKH";
  private static final String PUBKEY = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
  private static final String SIG = "CAFEBABE";

  private static final Script_input SCRIPTTXIN = new Script_input(TYPE, PUBKEY, SIG);

  @Test
  public void testGetType() {
    assertEquals(TYPE, SCRIPTTXIN.get_type());
  }

  @Test
  public void testGetSig() {
    assertEquals(SIG, SCRIPTTXIN.get_sig());
  }

  @Test
  public void testGetPub_key_recipient() {
    assertEquals(PUBKEY, SCRIPTTXIN.get_pubkey());
  }

  @Test
  public void testTestEquals() {
    assertEquals(SCRIPTTXIN, new Script_input(TYPE, PUBKEY, SIG));
    String random = "random";
    assertNotEquals(SCRIPTTXIN, new Script_input(random, PUBKEY, SIG));
    assertNotEquals(SCRIPTTXIN, new Script_input(TYPE, random, SIG));
    assertNotEquals(SCRIPTTXIN, new Script_input(TYPE, PUBKEY, random));
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/ScriptInputTest.kt
package com.github.dedis.popstellar.model.network.method.message.data.digitalcash

import com.github.dedis.popstellar.model.objects.security.PublicKey
import com.github.dedis.popstellar.model.objects.security.Signature
import org.junit.Assert
import org.junit.Test

class ScriptInputTest {
  @Test
  fun testGetType() {
    Assert.assertEquals(TYPE, SCRIPTTXIN.type)
  }

  @Test
  fun testGetSig() {
    Assert.assertEquals(Signature(SIG), SCRIPTTXIN.sig)
  }

  @Test
  fun testGetPub_key_recipient() {
    Assert.assertEquals(PUBKEY, SCRIPTTXIN.pubkey.encoded)
  }

  @Test
  fun testTestEquals() {
    Assert.assertEquals(SCRIPTTXIN, ScriptInput(TYPE, PublicKey(PUBKEY), Signature(SIG)))

    val random = "BBBBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB="

    Assert.assertNotEquals(SCRIPTTXIN, ScriptInput(random, PublicKey(PUBKEY), Signature(SIG)))
    Assert.assertNotEquals(SCRIPTTXIN, ScriptInput(TYPE, PublicKey(random), Signature(SIG)))
    Assert.assertNotEquals(SCRIPTTXIN, ScriptInput(TYPE, PublicKey(PUBKEY), Signature(random)))
  }

  companion object {
    private const val TYPE = "P2PKH"
    private const val PUBKEY = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
    private const val SIG = "CAFEBABE"
    private val SCRIPTTXIN = ScriptInput(TYPE, PublicKey(PUBKEY), Signature(SIG))
  }
}
